src/tfr_navigation/CMakeLists.txt

```diff
@@ -1,38 +1,64 @@
-# If you're having trouble getting your things to build, visit the ROS
-# tutorials at <http://wiki.ros.org/ROS/Tutorials> and also read the ROS
-# CMake documentation at <http://wiki.ros.org/catkin/CMakeLists.txt>.
-
 cmake_minimum_required(VERSION 2.8.3)
 project(tfr_navigation)
 
 add_compile_options(-std=c++11)
 
 find_package(catkin REQUIRED COMPONENTS
+  roscpp
+  tfr_msgs
+  geometry_msgs
+  nav_msgs
+  actionlib
 )
 
 find_package(GTest REQUIRED)
 
-# These are all for exporting to dependent packages/projects.
-# Uncomment each if the dependent project requires it
+include_directories(
+    include/${PROJECT_NAME}
+    ${catkin_INCLUDE_DIRS}
+    ${GTEST_INCLUDE_DIRS}
+)
+
 catkin_package(
-#  INCLUDE_DIRS include
-#  LIBRARIES tfr_sensor
-#  CATKIN_DEPENDS roscpp std_msgs
-#  DEPENDS system_lib
+    INCLUDE_DIRS include/${PROJECT_NAME}
+    LIBRARIES ${PROJECT_NAME} navigation_server navigation_client
+    CATKIN_DEPENDS  roscpp tfr_msgs actionlib nav_msgs geometry_msgs
 )
 
```

Comment:
Does `tfr_navigation` show up as a dependency for any other package?  If not, these lines in `catkin_package` are not required; this statement specifies what the package exports for other dependent packages.

Reply:
I had some issues with that line with that library that was marked as a library, I'll try deleting it, but I was not able to get my other code to build that reference external executables within the package without that list of dependencies

Reply:
Builds fine for me.  Give it another try?

I don't think having these lines in there will break anything, but it will slow down the compilation process a little.

Reply:
Fixed it! so in the future we fill this in only with the stuff that is going to be  used by external packages. I'll stick to that.